src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java

```diff
@@ -206,6 +207,7 @@ public KNNVectorFieldMapper build(BuilderContext context) {
                 );
                 if (knnMethodContext.getKnnEngine() == KNNEngine.LUCENE) {
                     log.debug(String.format("Use [LuceneFieldMapper] mapper for field [%s]", name));
+                    updateLuceneEngineStats(knnMethodContext.getKnnEngine());
```

Comment:
Why cant we put this in the lucene field mapper file?

Reply:
I want to give the update stat logic more visibility. Let me change it method that child class can overwrite, this will allow to kill two birds with one stone